File: mvvm-java/domain/src/test/java/com/wasisto/githubuserfinder/domain/usecases/GetSearchHistoryUseCaseTest.java
package com.wasisto.githubuserfinder.domain.usecases;

import com.wasisto.githubuserfinder.domain.interfaces.SearchHistoryDataSource;
import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.*;
import static org.mockito.Mockito.when;

public class GetSearchHistoryUseCaseTest {

    private GetSearchHistoryUseCase getSearchHistoryUseCase;

    @Mock
    private SearchHistoryDataSource searchHistoryDataSource;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        getSearchHistoryUseCase = new GetSearchHistoryUseCase(searchHistoryDataSource);
    }

    @Test
    public void testExecute() throws Exception {
        List<SearchHistoryItem> searchHistory = Arrays.asList(
                new SearchHistoryItem() {{
                    setId(1);
                    setQuery("john");
                }},
                new SearchHistoryItem() {{
                    setId(2);
                    setQuery("jane");
                }}
        );

        when(searchHistoryDataSource.getAll()).thenReturn(searchHistory);

        assertEquals(searchHistory, getSearchHistoryUseCase.execute());
    }
}

File: mvp-kotlin/domain/src/test/java/com/wasisto/githubuserfinder/domain/usecases/GetSearchHistoryUseCaseTest.kt
package com.wasisto.githubuserfinder.domain.usecases

import com.wasisto.githubuserfinder.domain.interfaces.SearchHistoryDataSource
import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem
import io.mockk.MockKAnnotations
import io.mockk.every
import io.mockk.impl.annotations.MockK
import org.junit.Before
import org.junit.Test

import org.junit.Assert.*

class GetSearchHistoryUseCaseTest {

    private lateinit var getSearchHistoryUseCase: GetSearchHistoryUseCase

    @MockK
    private lateinit var searchHistoryDataSource: SearchHistoryDataSource

    @Before
    fun setUp() {
        MockKAnnotations.init(this, relaxUnitFun = true)
        getSearchHistoryUseCase = GetSearchHistoryUseCase(searchHistoryDataSource)
    }

    @Test
    fun testInvoke() {
        val searchHistory = listOf(
            SearchHistoryItem(id = 1, query = "john"),
            SearchHistoryItem(id = 2, query = "jane")
        )

        every { searchHistoryDataSource.getAll() }.returns(searchHistory)

        assertEquals(searchHistory, getSearchHistoryUseCase())
    }
}